x-pack/platform/plugins/shared/cases/public/containers/api.ts

```diff
@@ -476,34 +477,75 @@ export const patchComment = async ({
 
 export const removeAlertFromComment = async ({
   caseId,
-  alertId: alertIdToRemove,
-  alertAttachment,
+  alertIds: alertIdsToRemove,
+  caseAttachments,
   signal,
 }: {
   caseId: string;
-  alertId: string;
-  alertAttachment: AlertAttachment;
+  alertIds: string[];
+  caseAttachments: AlertAttachmentUI[];
   signal?: AbortSignal;
-}): Promise<Attachment | void> => {
-  const { alertId, index, rule, version, id, owner } = alertAttachment;
-  if (Array.isArray(alertId) && Array.isArray(index) && alertId.length > 1) {
-    const alertIdx = alertId.indexOf(alertIdToRemove);
-    const newAlertId = [...alertId.slice(0, alertIdx), ...alertId.slice(alertIdx + 1)];
-    const newIndex = [...index.slice(0, alertIdx), ...index.slice(alertIdx + 1)];
-    return patchComment({
-      caseId,
-      commentId: id,
-      version,
-      patch: { type: AttachmentType.alert, alertId: newAlertId, index: newIndex, rule, owner },
-      signal,
-    });
-  } else {
-    return deleteComment({
-      caseId,
-      commentId: alertAttachment.id,
-      signal,
+}): Promise<PromiseSettledResult<Attachment | void>[]> => {
+  const attachmentMap = new Map<string, AlertAttachmentUI & { alertIndices: number[] }>();
+
+  alertIdsToRemove.forEach((alertIdToRemove) => {
+    let alertIndicesToRemove: number[] = [];
+    const alertAttachment = caseAttachments.find((attachment) => {
+      const idx = attachment.alertId.indexOf(alertIdToRemove.toString());
+      if (idx > -1) {
+        alertIndicesToRemove = [idx];
+        return attachment;
+      }
+      return undefined;
     });
-  }
+
+    if (!alertAttachment) {
+      throw new Error(`Alert with id ${alertIdsToRemove} not found in case ${caseId}`);
+    }
+
+    const existingAttachment = attachmentMap.get(alertAttachment.id);
+    if (existingAttachment) {
+      existingAttachment.alertIndices.push(...alertIndicesToRemove);
```

Comment:
This is going to be okay for the current size of indices to remove, but we've run into scenarios https://github.com/elastic/kibana/pull/212469 where an arbitrary spread to `.push()` actually broke the application at a significant enough argument size as there's an upper fn argument limit. We definitely won't hit that limit currently, but would prefer to future proof this a bit

Reply:
Just did, and looks really good, thanks! Just verifying the behavior we expect on the activity list with @paulewing , but will get back to you once I have that okay 👍🏾